velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -16,6 +16,126 @@
 
 #include "velox/substrait/SubstraitToVeloxExpr.h"
 #include "velox/substrait/TypeUtils.h"
+#include "velox/vector/FlatVector.h"
+
+using namespace facebook::velox;
+namespace {
+// Get values for the different supported types.
+template <typename T>
+T getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  VELOX_NYI();
+}
+
+template <>
+int8_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return static_cast<int8_t>(literal.i8());
+}
+
+template <>
+int16_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return static_cast<int16_t>(literal.i16());
+}
+
+template <>
+int32_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i32();
+}
+
+template <>
+int64_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i64();
+}
+
+template <>
+double getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.fp64();
+}
+
+template <>
+float getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.fp32();
+}
+
+template <>
+bool getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.boolean();
+}
+
+template <>
+uint32_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i32();
+}
+
+ArrayVectorPtr makeArrayVector(const VectorPtr& elements) {
+  BufferPtr offsets = allocateOffsets(1, elements->pool());
+  BufferPtr sizes = allocateOffsets(1, elements->pool());
+  sizes->asMutable<vector_size_t>()[0] = elements->size();
+
+  return std::make_shared<ArrayVector>(
+      elements->pool(),
+      ARRAY(elements->type()),
+      nullptr,
+      1,
+      offsets,
+      sizes,
+      elements);
+}
+
+ArrayVectorPtr makeEmptyArrayVector(memory::MemoryPool* pool) {
+  BufferPtr offsets = allocateOffsets(1, pool);
+  BufferPtr sizes = allocateOffsets(1, pool);
+  return std::make_shared<ArrayVector>(
+      pool, ARRAY(UNKNOWN()), nullptr, 1, offsets, sizes, nullptr);
+}
+template <typename T>
+void setLiteralValue(
+    const ::substrait::Expression::Literal& literal,
+    FlatVector<T>* vector,
+    vector_size_t index) {
+  if (literal.has_null()) {
+    vector->setNull(index, true);
+  } else if constexpr (std::is_same_v<T, StringView>) {
+    const auto typeCase = literal.literal_type_case();
+    if (literal.has_string()) {
+      vector->set(index, StringView(literal.string()));
+    } else if (literal.has_var_char()) {
+      vector->set(index, StringView(literal.var_char().value()));
+    } else {
+      VELOX_FAIL("Unexpected string literal");
+    }
+  } else if constexpr (std::is_same_v<T, Timestamp>) {
+    vector->set(index, Timestamp::fromMicros(literal.timestamp()));
+  } else if constexpr (std::is_same_v<T, Date>) {
+    vector->set(index, Date(literal.date()));
+  } else if constexpr (std::is_same_v<T, IntervalDayTime>) {
```

Comment:
Any particular reason not to implement `getLiteralValue<IntervalDayTime>` template?